### DISCUSSION TITLE
Archived: Name & Scope

### DISCUSSION THREAD

**Discussion by aiaa-deic** [aiaa-deic/computation-ai#4] [Polls]
I'm not sure if this makes sense to hold polls via discussions, but this is an option. There are a number of topics raised that deserve discussion regarding scope of the effort.

Candidate topics from the January SciTech discussions:

1. Computational Optimization Methods
2. Data-derived Surrogate Models
3. Large Language Models
4. Machine Perception
5. Machine Cognition / Neuromophic 
6. Probabilistic Methods & Uncertainty Quantification
7. Metamodeling
8. Generative Design
9. Inverse Design
10. Convergent / Set-based Concurrent Design
11. Co-simulation Methods & Architecture
12. Coupled (Multiphysics) Methods & Architecture
13. Multifidelity Methods
14. AI/ML + Mod/Sim Hybrid Methods
15. Provenance and Traceability / Reproducibility
16. Workflow / Automation
17. Data & Model Lifecycle Management
18. System Co-design (of hardware platform + software implementation + domain expertise)
19. Advanced automation for labor intensive / tedious activities
20. Systematic synthesis
21. Virtual commissioning
22. V&V processes toward virtual-before-physical 
23. [Decision provenance & knowledge stewardship](https://richardarthur.medium.com/provenance-for-decision-making-bf2c89d76ec2)
24. [Machine-augmented mindfulness](https://richardarthur.medium.com/machine-augmented-mindfulness-e844f9c54985)
25. [Modeling Maturity Assessment](https://richardarthur.medium.com/co-design-web-6f37664ac1e1)

**Comment by dahvid** [comment]
I liked the idea of focusing on a reference implementation, at least as a first step. I think this would naturally narrow or widen the scope as needed. There will no doubt be a lot of devils in the details which could be subjects for further investigation.

**Comment by aramdraper** [comment]
I think we should down-select and prioritize based on our ability to demonstrate the most success / provide the most value while expending the least amount of time/effort. While more nuanced and advanced topics would make for a more impressive example and paper, my interpretation of our objective is to provide generalized guidance and guidelines for digital engineering. Consequently, I think we should begin with more established topics that best represent a vast majority of practicing engineers.

**Reply by aramdraper** [reply]
Now, each and every one of us could have differing opinions on what those "established" topics should be, but I will put in my 2 cents - we should focus on **Probabilistic Methods & Uncertainty Quantification**, **System Co-design**, and **AI/ML + Mod/Sim Hybrid Methods**. We can come up with a basic example where we generate a virtual system and the physical system simultaneously. The Mod/Sim methods represent system behavior in real time. AI/ML can be used for some sort of prognostication based on current events. The use of probabilistic methods and uncertainty quantification in this example can come from instrument (aleatoric) uncertainty.

**Reply by arthurrge** [reply]
Based on this, a refined list:

**Prioritized-up**:

- **Probabilistic Methods & Uncertainty Quantification**
- **System Co-design (of hardware platform + software implementation + domain expertise)**
- **AI/ML + Mod/Sim Hybrid Methods**

**Suggest discussion**:

- **Large Language Models** (because this is receiving tremendous attention currently)
- Data-derived Surrogate Models (in context of AI policy discussions)
- Multifidelity Methods (has been noted a priority by AFRL)

**Prioritized-down**:

AI-oriented

- Machine Perception
- Machine Cognition / Neuromophic
- Advanced automation for labor intensive / tedious activities

Methods

- Metamodeling
- Computational Optimization Methods
- Generative Design (in the sense of automated search of solution space)
- Inverse Design
- Convergent / Set-based Concurrent Design
- Co-simulation Methods & Architecture
- Coupled (Multiphysics) Methods & Architecture
- Systematic synthesis

Supporting (perhaps in scope for Ecosystem team?)

- Provenance and Traceability / Reproducibility
- Workflow / Automation
- Data & Model Lifecycle Management
- Virtual commissioning
- V&V processes toward virtual-before-physical
- [Decision provenance & knowledge stewardship](https://richardarthur.medium.com/provenance-for-decision-making-bf2c89d76ec2)
- [Machine-augmented mindfulness](https://richardarthur.medium.com/machine-augmented-mindfulness-e844f9c54985)
- [Modeling Maturity Assessment](https://richardarthur.medium.com/co-design-web-6f37664ac1e1)

**Reply by coleyc** [reply]
In my opinion, from your Prioritized Up and Suggest discussion items, it seems to me the following topics are extremely relevant/important and should be a focus of our effort:

System Co-design (of hardware platform + software implementation + domain expertise)
AI/ML + Mod/Sim Hybrid Methods
Data-derived Surrogate Models (in context of AI policy discussions)
Multifidelity Methods (has been noted a priority by AFRL)

**Comment by arthurrge** [comment]
### Computing Governance / Practices (Private Datacenter / Public Cloud / Hybrids)

A side conversation at SciTech raised a potential additional topic that is likely shared and significant in this vein:
In-house vs. Cloud HPC / Computational Infrastructure - in particular the governance around appropriate use, assessing benefits vs. cost/risks, and some potential capabilities (e.g., collaboration through public cloud) and gaps (e.g., IP vs. collaboration) -- and leverage existing guidelines already emerging in the DEIC Ecosystem effort and outside-AIAA sources.

**Comment by arthurrge** [comment]
Pulling in the governance topic as an additional bullet - here is the scope as presented Jan 9th

### SciTech 2024 DEIC Face-to-Face Scoping Computing/AI

**(A) Advancing Computational Methods for Aerospace Science & Engineering(Architectures and frameworks for application of novel methods and capabilities)**
- Enabling Multi-X (-scale, -fidelity, -physics/co-simulation/coupled, -disciplinary/transdisciplinary design + manufacturing + sustainment) modeling & analyses
- AI-Hybrid Methods (AI/ML + Mod/Sim), (fusing synthetic + measured data), (formulaic-models + data-derived models/surrogates)
- System Co-design (of hardware platform + software implementation + domain expertise), potentially leveraging emerging capabilities in AI/LLMs ([ChatGPT](https://www.engineering.com/story/chatgpt-is-pushing-engineering-buttons)/[Copilot](https://adoption.microsoft.com/en-us/copilot/)) toward composability and optimization in such multidimensional trade-off spaces.
- Generative & Inverse Methods (spec desired effect, derive candidate causes), (trade-off-space mapping), etc.

**(B) Framework for Adoption (of these Advancing Computational Methods for Aerospace Science & Engineering )**
- [Modeling Maturity Assessment](https://richardarthur.medium.com/co-design-web-6f37664ac1e1) (physics-basis, data-derivation quality, predictive accuracy, bounded confidence, numerical robustness, software engineering quality, etc.)
- Uncertainty Quantification (for data-derived models/probabilistic methods, bounding region of model competence, data requirements)
- Verification & Validation (Traceability, Reproducibility, Explainability, & Provenance) - including tracking and retiring [Intellectual Debt](https://medium.com/berkman-klein-center/from-technical-debt-to-intellectual-debt-in-ai-e05ac56a502c)
- Certification (gaps, risks, & grand challenges) + (impact, value, & opportunities)
- Governance (guidelines and oversight for applying technologies / infrastructure – internal investments vs. external services)

**Examples of important factors to ensure are included:**
- Consistency for commercial and military domains
- Digital Thread as conduit of data to/from/between tools & methods
- Enterprise architecture implications (ERP, PLM, MES, MRO, etc.)
- Full lifecycle (design + manufacture + operation + sustainment)
- Modes of data (unstructured text, structured data, boundary conditions, geometry, tolerances, visual)
- Implications for both data scientists and engineers for lifecycle management (of data, models, tools) to succeed, comply, innovate upon

**Comment by arthurrge** [comment]
Alicia Kim and I did a mini brainstorming on the scope and practical next steps.
Here is an attempt at capturing the thinking...

1. The path of this subcommittee may not follow the path of the others (position paper of definition, then one for use) -- because this is more about proficiently employing technologies and techniques to all the other DEIC stuff. But we'll see how it unfolds.
2. Let's first continue the scoping discussion by way of collecting together a survey of use cases (in particular barrier problems / grand challenges limiting leaps forward in state of the art in digital/virtual modeling & simulation).  This could include a wider RFI to help assess the untapped problem space we would like to unlock.
3. Then let's try to categorize them - I think a default baseline for this should be the NASA 2040 framework - which I think is awesome - and then we can assess any deltas from there. I hope Steve Arnold can assist here.
4. Additionally - survey a collection of methods, libraries, software that we consider core to the supporting toolchain for computationally supporting digital engineering practices.
5. Additionally - survey a collection of maturity assessments and guidelines from other (mature?) sources like NAFEMS, INCOSE, ASME, etc.
6. Publish findings consolidating all of the above, with a proposal for some options toward collaborations that might address these barriers through combination of experts from the camps of (A) Domain SMEs + (B) Hardware Experts/Architects + (C) Implementation / RSE / Performance and Scalability / Methods Experts.

We can expand on some of the nuance here, I'm just working to capture the main points before post-conference amnesia sets in.  The gist is we hypothesize artificial obstacles due to ignorance that can be overcome by bringing (A, B, C) expertise to the same table to tight-loop work through trade-space to discover opportunities in light of the rapid advances in computing/methods - including AI.

**Reply by aramdraper** [reply]
All, I reflected these points in the to-do for the first meeting in support of generating a ToC for the position paper!

**Comment by arthurrge** [comment]
Here is a draft update to the one-slide summary post-SciTech2024. 
Feedback always appreciated!
[CAEHPCAI-post-SciTech2024.pdf](https://github.com/aiaa-deic/computation-ai/files/13964235/CAEHPCAI-post-SciTech2024.pdf)

**Comment by digitalhangarguy** [comment]
Does this group have any participation from AIAA's Non-Deterministic Approaches TC?

**Comment by arthurrge** [comment]
### (SENT TO AIAA DEIC LEADS)
Fresh out of the monthly (Computational Methods/AI Subcommittee) call, I wanted to cycle with this group on a few points, seeking feedback and guidance.
 

1. Problem: The focus of this subcommittee remains a challenge – even in crafting a clear-yet-correct name. 
   - This challenge then cascades into delaying progress on the activities (including proposing a technical panel or workshop for SciTech 2025, though we agree we should have one.)
   - One observation was that other subcommittees center on a Digital Engineering “thing” – twins, thread, models, even ecosystem, while the scope for this group concerns the advancing of the methods that underlie all those things in the other subcommittees.
   - It is the subcommittee’s perception there are several significant technical topics which appear to constitute AIAA gaps (or at least for DEIC), most prominently: (a) HPC-driven methods + Co-designed solutions (expertise in subject-matter + software/implementation + hardware/architecture) & (b) AI/Human-Machine Collaboration/ Advanced Automation
2.  Request: Outside clarification / assumption verification to assist in down-scoping or amplifying. 
   - AI/ML is an attention lightning rod at the moment, and (probably correctly) not explicitly its own TC/IC/etc. AI is an importantly enabling computational method, but AI is (merely) one of many methods advancing Digital Engineering.
   - Assumption: AI / Human-Machine Collaboration can be implicit in “Computational Methods for Digital Engineering”.
   - The AIAA [CFD 2030](https://www.cfd2030.com/) (e.g.,  [Fact Sheet](https://www.cfd2030.com/CFD-2030-Fact-Sheet-2022-final.pdf)) Vision Elements cover a lot of the target ground already – but are not extended beyond CFD, leaving application gaps such as for materials.
   - Uncertainty: how to complement CFD 2030 w/o redundancy, contradiction, over-dependency while extending appropriately.
   - Similarly, the DEIC efforts aimed at Digital Ecosystem, Digital System Model, and Confidence in Models could cover how advancing computational methods impact:
                                                               i.      Data mode integration (tabular/structured, time-series, geometry/tolerances, visual, text/unstructured, machine-readability, training sets, …) & [FAIR](https://www.go-fair.org/) conformity
                                                             ii.      Exo-engineering enterprise integration (ERP, MES, MRO, PHM, etc.)
                                                           iii.      Model competency ([VVUQ](https://event.asme.org/VandV), [applicability & provenance metadata](https://www.nafems.org/publications/resource_center/assess_esms/), UX/productivity, lifecycle management, etc.)
                                                            iv.      Organizational [modeling literacy](https://drive.google.com/file/d/1DhkTct_4D5AElMoDeSOXsmBUISNtxdu1/view?usp=sharing) & culture
Seeking clarification/confirmation: (i-iv covered and) value of focus on computational advances impacting state-of-the-art model fidelity & realism (singly or through integration/coupling – i.e., multiscale, multifidelity, multidisciplinary, multiphysics)
  
Rick & Abhishek
AIAA DEIC CAE/? Subcommittee

**Reply by arthurrge** [reply]
### Reply to this from Phil Zimmerman (now at SERC/Stevens):
(I have an opinion on everything, and share most of them!)

As the least experienced with DEIC goings-on, here's the few thoughts i can offer: 

Regarding Assumption: AI / Human-Machine Collaboration can be implicit in “Computational Methods for Digital Engineering”.

I prefer to think about digital engineering as foundational to adopting computational methods in doing a task.  Computational methods themselves, are not the point of digital engineering. In the end, it's the application of those computational methods that's important.

So, a question: is AI/Human Machine collaboration a computational method, or an end state of adopting computational methods?  I don't know the answer, and I'm not likely to be convinced either way....  Maybe it's both, and a whitepaper should address it as such.

OR.....A shift to computational methods is necessary condition to being able to fully utilize AI/Human Machine collaboration?  Why?  What happens if you don't embrace computational methods

**Reply by aramdraper** [reply]
"In the end, its the application of those computational methods that's important" --> maybe we could take a step back and examine the application of different kinds of computational methods and computing approaches that some may employ. E.g.,

- Types of Computational Models/Methods
   - Nondeterministic Approaches (e.g., AI)
      - Machine Learning (except algorithms with fixed inputs and outputs)
      - Deep Learning
      - Natural Language Processing
      - Genetic Algorithms
      - Monte Carlo Simulation
      - Stochastic Modeling
   - Deterministic Approaches
      - Finite Element Method
      - Computational Fluid Dynamics
      - Any Algorithms That Would Bring the Same Results Each Iteration
 -High Performance Computing
    - Parallel Computing
    - Cluster Computing
    - Distributed Computing

Depending on if you're dealing with nondeterministic and deterministic approaches, the way you conduct digital engineering would change. Additionally, depending on specific algorithms employed, the way you would conduct digital engineering would change. Another dimension is whether you have access to high performance computing, in which case, the structure of your digital tools would also be liable to change. We could probably spend a very long time to come up with these comparisons, and maybe that should be the focus of the paper.

**Reply by arthurrge** [reply]
Sharing here a comment I just cycled with the DOE leadership computing folks, based on a point Brian van Essen made at SOS, regarding AI vs. traditional architectural trajectories (CSE apps can no longer rely upon leveraging the momentum of GPUs)
1.	Hyperscalers are focused on learning from human data in text or text-labeled visuals
2.	The architectures to do this optimize for cost/energy by pushing down numerical precision
3.	For scientific ML, the underlying data – and more importantly the data needed to fill gaps in unknowns via synthetic data inherently require high precision, which is wholly counter to the hyperscaler architecture directions
4.	Therefore, these two classes of architectures must co-exist and complement each other – but will not be served solely by the commercial side
5.	The question may then come down to how much of the low-precision/AI side of this should be taken on at the LCFs vs. shouldered by cloud

**Comment by aramdraper** [comment]
All,

Given the aggressive schedule for publication that we are putting ourselves under, I am proposing that we specify a topic for us to focus on. I was thinking more about our potential name and what we would focus on. I have come to realize that trying to create a single position paper on applying deterministic methods, non-deterministic methods, high performance computing, and other methods in the same paper would be far too challenging to fit into a reasonable paper size, would time-constrain us from giving each topic its requisite time, and would probably be infeasible for Sci-Tech. I think instead we should focus on each of these computational design and analysis method groups one at a time and drill down on the chapters/topics that we have already talked about (see attached, Slides 8 and 9.)
[DEIC-CAE Meeting [Updated] (04-02-24).pptx](https://github.com/aiaa-deic/computation-ai/files/14948447/DEIC-CAE.Meeting.Updated.04-02-24.pptx)

In terms of a name, I propose: **Computational Design and Analysis for Digital Engineering (CoDADE.)** It addresses the fact that our focus is on how computational methods are applied in digital engineering and addresses all the possible topics we might address.

We cannot afford to spin our tires anymore. If no one has strong objections, then we should continue with these. Let us start with non-deterministic methods (ML, NLP, Stochastic Methods, etc.) because they are a hot topic and are not nearly mature as deterministic methods (FEM, CFD, etc.)

Thoughts? I will assume no response = consensus and we will finalize by the end of next week.

**Comment by aramdraper** [comment]
Hello everyone,

I hope this message finds you well. It has been a couple weeks since I [proposed](https://github.com/aiaa-deic/computation-ai/discussions/4#discussioncomment-9086258) the subcommittee name: Computational Design and Analysis for Digital Engineering (CoDADE) and suggested that we should ultimately create two position papers based on the subject areas of “deterministic methods” and “non-deterministic methods” respectively. As high-performance computing can imply different things to these subject areas, I propose that being a topic of study in both subject area position papers. As no one has raised any objections in the last two weeks, I will assume consensus with the proposal. With that in mind, I think our first position paper should address “non-deterministic” methods (e.g., ML, NLP, Stochastic Methods) because they are collectively a hot topic right now and are not mature as our “deterministic” methods (e.g., FEM, CFD.) Also, as many of these non-deterministic methods are becoming tools to supplement the deterministic methods, I believe it is critical we start with these “non-deterministic” methods. 

Because of how widespread the use of methods like ML, NLP, etc. are, I think we should start by sending out surveys involving what these methods are being used for / how they are being used. With this knowledge, we can begin to assess the impacts that these methods have on the different chapter areas we had originally discussed (**see below.) For the next subcommittee meeting (04-07-2024), I would like us to have several questions on each of these chapter areas drafted. As we want surveys on each chapter area to be less than 10 minutes each, we want to draft 10-15 solid questions.

ACTION ITEM: Draft survey questions that match up to the **chapter areas below. Need 10-15 per chapter area for surveys.

Reach out if you have any additional questions. Thank you very much, and I look forward to our meeting in a couple weeks!

Best Regards,

Abhishek Ram [Personal Cell: (508)-918-9885]

**
1.	Computational Design and Analysis Framework Development
a.	Determining what kind of model you need for an application (e.g., Multi-X, AI-Hybrid, Other Non-deterministic Approaches)
b.	Model validation approaches (including applicability of previous studies)
c.	Maturity assessment for different types of models
d.	Uncertainty Quantification based on models used for application
e.	Simulation requirements driven by the kind of model needed and model validation
2.	Impacts of Computational Design and Analysis on Enterprise Data and Software Architecture
a.	Modes and states of data (e.g., unstructured text, structured data, boundary conditions, geometry, tolerances, visual)
b.	Software verification / quality assurance
c.	Enterprise architecture implications (ERP, PLM, MES, MRO, etc.)
d.	Data and software governance
3.	System Co-Design
a.	Cyber-physical tradeoffs based on mod/sim, software, and data requirements (e.g., impact of HPC, different types of models)
b.	Barriers of scale / extensibility
c.	Life cycle management across full life cycle
4.	Impacts of Socio-Technical Interactions on Computational Design and Analysis Strategy
a.	Use of semantic technologies for supporting human-machine interactions
b.	Utilitarian-driven Computational Design and Analysis requirements generation
i.	Modeling organizational culture and decision-making and impact on mod/sim requirements generation
ii.	Accounting for impacts of interactions of multiple teams / disciplines in mod/sim requirements generation

**Comment by arthurrge** [comment]
Archival
-
I am going to archive this particular thread and create a fresh one, as I believe we need to step back to move forward. There are numerous important topics contained in this thread - but I believe those ambitions extend beyond our reach and are somewhat conflated with the scope of other DEIC efforts.

See also: <[Scope re: DECO](https://github.com/aiaa-deic/computation-ai/discussions/3#discussioncomment-9287676)>, <[Scope re: DevSecOps/Trust](https://github.com/aiaa-deic/computation-ai/discussions/3#discussioncomment-9287706)>

Purpose
-
In fact, the purpose of this group has been to "catch" a few important gap topics that have not been otherwise addressed in the AIAA DEIC working groups. This has proven very challenging to scope succinctly through a consensus-driven approach.  That process has, however, populated a rich landscape of potential topics to be addressed (whether by this effort or otherwise within DEIC - @opinonfischer & @jfmatlik take note!)

1. At the root of the themes discussed so far, we acknowledge transformational advances in computational (digital / virtualized) methods applied to science and engineering for aeronautics and astronautics. (Such as):
    - AI/ML methods and data-derived models (including surrogates, generative, inverse, etc.)
    - HPC/advanced computing (see CFD 2030 bullet below)
    - Greatly-expanded problem scope based on scalable/affordable infrastructure
2. The AIAA [CFD 2030](https://www.cfd2030.com/) (e.g., [Fact Sheet](https://www.cfd2030.com/CFD-2030-Fact-Sheet-2022-final.pdf)) Vision Elements cover a lot of the target ground already
    - physics-based, predictive modeling (which can/should extend to co-simulation and coupled physics)
    - quantification of errors and uncertainties arising from physical modeling, mesh and discretization, etc.
    - automation in all steps of the analysis process: geometry creation, meshing, large databases of simulation results, extraction and understanding of simulation information, etc.
    - impact/opportunity from emerging HPC architecture advances, including: memory hierarchies (latencies/bandwidths), programming paradigms and runtime environments, etc.
 
    but are not extended beyond CFD, leaving application gaps such as for materials modeling, part lifing, etc.
3. Therefore, we should confine scope to technologies and practices that are unique and urgently-applicable for aero/defense science and engineering, leveraging and expanding upon adjacent efforts and synergies (CFD, DECO, NAFEMS, ASME, SERC, ECP, etc.) toward valuable guidance in shared efforts to advance the state of the art in methods and practices for this technical community.

(Looking to active participants to take note / contribute in the new thread - @aramdraper @Papaghena @coleyc @NigeTaylo @smarnold59)
